### ISSUE TITLE
PauliTrotterEvolution fails on certain combinations of pauli operators

### ISSUE THREAD

**Issue by hay-k** [Qiskit/qiskit-terra#5733]
### Information

- **Qiskit Aqua version**: 0.8.0
- **Python version**: 3.7.4

### What is the current behavior?

`PauliTrotterEvolution` exits with error when asked to convert operators of certain structure. Here is the error message

```
Traceback (most recent call last):
  File "D:/qiskit advocate/test_project/main.py", line 3, in <module>
    PauliTrotterEvolution(Suzuki(order=1, reps=1)).convert(o.exp_i())
  File "D:\python_venvs\qiskit_venv\lib\site-packages\qiskit\aqua\operators\evolutions\pauli_trotter_evolution.py", line 100, in convert
    return self._recursive_convert(operator)
  File "D:\python_venvs\qiskit_venv\lib\site-packages\qiskit\aqua\operators\evolutions\pauli_trotter_evolution.py", line 130, in _recursive_convert
    circuit_no_identities.primitive.global_phase = -sum(identity_phases)  # type: ignore
AttributeError: 'ComposedOp' object has no attribute 'primitive'
```

### Steps to reproduce the problem

The following code snippet will reproduce the error message above

```
from qiskit.aqua.operators import *
o = ((I + Z) ^ I) + ((I - Z) ^ X)
PauliTrotterEvolution(Suzuki(order=1, reps=1)).convert(o.exp_i())
```

### What is the expected behavior?
The code snippet above is expected to run without errors and produce result. Even though I am doing something weird (exponentiating the `2*CNOT` operator (don't ask me why :D)), the operator being exponentiated is hermitian, and I believe it is ok to exponentiate it and expect a meaningful result.

If I do the same thing using matrix operators
```
Imat = I.to_matrix_op()
Xmat = X.to_matrix_op()
Zmat = Z.to_matrix_op()
omat = ((Imat + Zmat) ^ Imat) + ((Imat - Zmat) ^ Xmat)
omat_evo = MatrixEvolution().convert(omat.exp_i())
```
the code runs without errors, and calculates the expected result. Indeed, the result is in line with `scipy` exponentiation:
```
from scipy.linalg import expm
omat_evo_expected = expm(-2j * np.array([[1, 0, 0, 0],
                                    [0, 1, 0, 0],
                                    [0, 0, 0, 1],
                                    [0, 0, 1, 0]]))
print(np.allclose(omat_evo.to_matrix(), omat_evo_expected))
```
this prints `True`.

### Suggested solutions
It looks like something goes wrong in the recursive calls of `_recursive_convert` in `PauliTrotterEvolution`, but have not yet had a chance to investigate in detail.

**Comment by jlapeyre** [comment]
As of at least 0c8bb3d, the example in the opening comment now results in a `RecursionError`.

**Comment by jlapeyre** [comment]
I tested the example in the original comment above with recent master 302f60abe. That is, the following
```python
from qiskit.aqua.operators import *
o = ((I + Z) ^ I) + ((I - Z) ^ X)
PauliTrotterEvolution(Suzuki(order=1, reps=1)).convert(o.exp_i())
```

I find that it no longer errors, and furthermore reproduces the same result as the matrix example above. This was very likely fixed by PR #6111 in response to issue #6099

It looks like this issue is resolved and can be closed.
